lib/libc/arcmwdt/libc-hooks.c

```diff
@@ -0,0 +1,113 @@
+/* stdout_console.c */
+
+/*
+ * Copyright (c) 2014, 2021 Wind River Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <sys/libc-hooks.h>
+#include <syscall_handler.h>
+#include <string.h>
+
+static int _stdout_hook_default(int c)
+{
+	(void)(c);  /* Prevent warning about unused argument */
+
+	return EOF;
+}
+
+static int (*_stdout_hook)(int) = _stdout_hook_default;
+
+void __stdout_hook_install(int (*hook)(int))
+{
+	_stdout_hook = hook;
+}
+
+int z_impl_zephyr_fputc(int c, FILE *stream)
```

Comment:
Here you are implicitly implementing the "minimal libc" hook interface functions:
https://github.com/zephyrproject-rtos/zephyr/blob/775cf55d9268201be6f42523c80ffad808fd3209/include/sys/libc-hooks.h#L19-L37

You should add `#elif defined(CONFIG_ARCMWDT_LIBC)` there and add only the required hook functions for the ARC MWDT libc.

Reply:
Fixed in new implementation, thanks!